### PR TITLE
Improve format code script

### PR DESCRIPTION
**Pull request by rnestler**
Fixes the issues reported by the shellcheck utility.

### CODE REVIEW

File: dev/format_code.sh [hunk 1]
```diff
@@ -78,8 +81,7 @@ do
   done
 done
 
-echo "Finished: $COUNTER files modified."
-
+echo "Finished: $COUNTER files modified."
```
Comment (ubruhin) [inline dev/format_code.sh]: Hmm what happened here? ;)

File: dev/format_code.sh [hunk 2]
```diff
@@ -34,13 +36,14 @@ if [ "$DOCKER" == "--docker" ]; then
 
   if [ "$(docker images -q $DOCKER_IMAGE | wc -l)" == "0" ]; then
     echo "Building clang-format container..."
-    cd "$REPO_ROOT/dev/clang-format"
-    docker build . -t librepcb/clang-format:6
-    cd -
+    (
+      cd "$REPO_ROOT/dev/clang-format"
+      docker build . -t librepcb/clang-format:6
+    )
```
Comment (ubruhin) [inline dev/format_code.sh]: An alternative might be to remove the `cd` and pass the path to `docker build` instead...

Comment (rnestler) [reply dev/format_code.sh]: That would indeed be simpler and cleaner.

File: dev/format_code.sh (hunk 1 above)
Comment (rnestler) [reply dev/format_code.sh]: Huh no idea. I normally look at the git diff before commiting... Fixed.

File: dev/format_code.sh (hunk 2 above)
Comment (rnestler) [reply dev/format_code.sh]: Fixed.

**Comment by rnestler** [comment]
> Would you also be interested in fixing a bug? :grin: The script fails if files were removed (compared to master) because git diff also returns removed files, and those can't be found by clang-format...

Probably wouldn't be too hard. But not in this PR :wink:

**Comment by ubruhin** [comment]
OK :)

Thanks for the fixup, looks good :+1:

**Pull request merged**